### DISCUSSION TITLE
Homeassistant Add-On mit HMIP-RFUSB unter Proxmox funktioniert nicht

### DISCUSSION THREAD

**Discussion by philippsandhaus** [OpenCCU/OpenCCU#2158] [🇩🇪 Fragen&Antworten]
Hallo zusammen,

ich habe jetzt einiges probiert und komme leider nicht weiter, wäre für jeden Hinweis dankbar. Zu meiner Situation: Ich möchte gerne von einem ELV-Charly mit RaspberryMatic migrieren zum RaspberyyMatic AddOn in meiner Homeassistant Installation. Diese ist installiert als HomeAssistant OS 9.4 mit HomeAssistant 2023.1.2 auf einem kleinen Dell-Server innerhalb einer Proxmox-Umgebung. Hierüber habe ich auch andere USB-Geräte wie z.B. einen Zigbee-Stick erfolgreich an HomeAssistant durchgereicht. Das gleich habe ich auch mit einem HMIP-RFUSB gemacht, der grundsätzlich auch vom RaspberryMatic-Addon erkannt zu werden. Nachdem ich nach der Anleitung zur Migration hier im Wiki vorgegangen bin und ich Probleme bei der Erreichbarkeit meiner Geräte hatte, teste ich gerade mit einer frischen Installation ohne Einspielen des alten Backups. Trotzdem habe ich die gleichen Probleme:

- Im Webfrontend äussert sich das darin, dass ich zwar den Anlernmodues sowohl für Hm als auch HmIP starten kann, Geräte aber nicht gefunden werden. Ausserdem wird in regelmässigen Abständen die Alarmmeldung "WatchDog: hmipserver-restart" angezeigt
- Das Debug-Log sieht folgendermassen aus:
```
***** messages *****
Jan 12 21:39:05 de838cd8-raspberrymatic syslog.info syslogd started: BusyBox v1.35.0
Jan 12 21:39:05 de838cd8-raspberrymatic user.info firewall: Warning: Extension limit revision 0 not supported, missing kernel module? ip6tables: No chain/target/match by that name.
Jan 12 21:39:05 de838cd8-raspberrymatic user.info firewall: Warning: Extension hl revision 0 not supported, missing kernel module? ip6tables: No chain/target/match by that name.
Jan 12 21:39:05 de838cd8-raspberrymatic user.info firewall: configuration set
Jan 12 21:39:09 de838cd8-raspberrymatic daemon.err xinetd[950]: Unable to read included directory: /etc/config/xinetd.d [file=/etc/xinetd.conf] [line=14]
Jan 12 21:39:09 de838cd8-raspberrymatic daemon.crit xinetd[950]: 950 {init_services} no services. Exiting...
Jan 12 21:39:09 de838cd8-raspberrymatic user.info root: Updating RF Lan Gateway Coprocessor Firmware
Jan 12 21:39:09 de838cd8-raspberrymatic user.debug update-coprocessor: firmware filename is: coprocessor_update_hm_only.eq3
Jan 12 21:39:09 de838cd8-raspberrymatic user.info root: Updating RF Lan Gateway Firmware
Jan 12 21:39:09 de838cd8-raspberrymatic user.info update-lgw-firmware: No gateway found in config file /etc/config/rfd.conf
Jan 12 21:44:19 de838cd8-raspberrymatic daemon.info : starting pid 1469, tty '': '/bin/mv /tmp/boot.log /var/log/boot.log'
Jan 12 21:44:19 de838cd8-raspberrymatic daemon.info : starting pid 1470, tty '/dev/null': '/usr/bin/monit -Ic /etc/monitrc'
Jan 12 21:44:19 de838cd8-raspberrymatic user.info monit[1470]: Starting Monit 5.30.0 daemon with http interface at /var/run/monit.sock
Jan 12 21:44:19 de838cd8-raspberrymatic user.info monit[1470]: 'de838cd8-raspberrymatic' Monit 5.30.0 started
Jan 12 21:44:24 de838cd8-raspberrymatic local0.err ReGaHss: ERROR: XmlRpc: Error in XmlRpcClient::writeRequest: write error (error 111). [error():iseXmlRpc.h:281]
Jan 12 21:44:24 de838cd8-raspberrymatic local0.warn ReGaHss: WARNING: XMLRPC 'init': rpcClient.execute() failed (url: xmlrpc://127.0.0.1:39292/groups, params: {"xmlrpc_bin://127.0.0.1:31999","1008"}, result: nil) [CallXmlrpcMethod():iseXmlRpc.cpp:2602]
Jan 12 21:44:24 de838cd8-raspberrymatic local0.err ReGaHss: ERROR: XMLRPC 'init' call failed (interface: 1008, params: {"xmlrpc_bin://127.0.0.1:31999","1008"}) [CallInit():iseXmlRpc.cpp:1204]
Jan 12 21:44:24 de838cd8-raspberrymatic local0.err ReGaHss: ERROR: failed CallInit() for interface=VirtualDevices [ThreadFunction():iseRTHss.cpp:163]
Jan 12 21:44:24 de838cd8-raspberrymatic local0.err ReGaHss: ERROR: XmlRpc: Error in XmlRpcClient::writeRequest: write error (error 111). [error():iseXmlRpc.h:281]
Jan 12 21:44:24 de838cd8-raspberrymatic local0.warn ReGaHss: WARNING: XMLRPC 'init': rpcClient.execute() failed (url: xmlrpc://127.0.0.1:32010, params: {"xmlrpc_bin://127.0.0.1:31999","1009"}, result: nil) [CallXmlrpcMethod():iseXmlRpc.cpp:2602]
Jan 12 21:44:24 de838cd8-raspberrymatic local0.err ReGaHss: ERROR: XMLRPC 'init' call failed (interface: 1009, params: {"xmlrpc_bin://127.0.0.1:31999","1009"}) [CallInit():iseXmlRpc.cpp:1204]
Jan 12 21:44:24 de838cd8-raspberrymatic local0.err ReGaHss: ERROR: failed CallInit() for interface=HmIP-RF [ThreadFunction():iseRTHss.cpp:163]
Jan 12 21:44:34 de838cd8-raspberrymatic user.err monit[1470]: 'udevdEnabled' status failed (0) -- no output
Jan 12 21:44:35 de838cd8-raspberrymatic user.err monit[1470]: 'sshdEnabled' status failed (1) -- no output
Jan 12 21:44:35 de838cd8-raspberrymatic user.err monit[1470]: 'hs485dEnabled' status failed (2) -- grep: /var/etc/hs485d.conf: No such file or directory
Jan 12 21:44:35 de838cd8-raspberrymatic user.err monit[1470]: 'hmlangwEnabled' status failed (1) -- no output
Jan 12 21:44:36 de838cd8-raspberrymatic user.warn monit[1470]: 'HMIPServer' failed protocol test [DEFAULT] at [localhost]:39292 [TCP/IP] -- Connection refused
Jan 12 21:44:36 de838cd8-raspberrymatic user.warn monit[1470]: 'hasUSB' status failed (1) -- no output
Jan 12 21:44:36 de838cd8-raspberrymatic user.err monit[1470]: 'hb_rf_eth-CheckEnabled' status failed (1) -- no output
Jan 12 21:44:36 de838cd8-raspberrymatic user.warn monit[1470]: 'rpi4usb3Check' status failed (1) -- no output
Jan 12 21:44:36 de838cd8-raspberrymatic user.err monit[1470]: 'tailscaleEnabled' status failed (1) -- no output
Jan 12 21:44:36 de838cd8-raspberrymatic user.err monit[1470]: Lookup for '/media/usb1' filesystem failed  -- not found in /proc/self/mounts
Jan 12 21:44:36 de838cd8-raspberrymatic user.err monit[1470]: Filesystem '/media/usb1' not mounted
Jan 12 21:44:36 de838cd8-raspberrymatic user.err monit[1470]: 'usb1' unable to read filesystem '/media/usb1' state
Jan 12 21:44:36 de838cd8-raspberrymatic user.info monit[1470]: 'usb1' trying to restart
Jan 12 21:44:50 de838cd8-raspberrymatic user.warn monit[1470]: 'HMIPServer' failed protocol test [DEFAULT] at [localhost]:39292 [TCP/IP] -- Connection refused
Jan 12 21:44:51 de838cd8-raspberrymatic user.warn monit[1470]: 'hasUSB' status failed (1) -- no output
Jan 12 21:44:52 de838cd8-raspberrymatic user.warn monit[1470]: 'rpi4usb3Check' status failed (1) -- no output
Jan 12 21:44:52 de838cd8-raspberrymatic user.err monit[1470]: Filesystem '/media/usb1' not mounted
Jan 12 21:44:52 de838cd8-raspberrymatic user.err monit[1470]: 'usb1' unable to read filesystem '/media/usb1' state
Jan 12 21:44:52 de838cd8-raspberrymatic user.info monit[1470]: 'usb1' trying to restart
Jan 12 21:45:00 de838cd8-raspberrymatic local0.err ReGaHss: ERROR: XmlRpc: Error in XmlRpcClient::writeRequest: write error (error 111). [error():iseXmlRpc.h:281]
Jan 12 21:45:00 de838cd8-raspberrymatic local0.warn ReGaHss: WARNING: XMLRPC 'getValue': rpcClient.execute() failed (url: xmlrpc://127.0.0.1:32010, params: {"00041F2995E72C:0","CARRIER_SENSE_LEVEL"}, result: nil) [CallXmlrpcMethod():iseXmlRpc.cpp:2602]
Jan 12 21:45:00 de838cd8-raspberrymatic local0.err ReGaHss: ERROR: XMLRPC 'getValue' call failed (interface: 1009, params: {"00041F2995E72C:0","CARRIER_SENSE_LEVEL"}) [CallGetValue():iseXmlRpc.cpp:1435]
Jan 12 21:45:00 de838cd8-raspberrymatic local0.err ReGaHss: ERROR: CallGetValue failed; sVal = 0.000000 [ReadValue():iseDOMdpHSS.cpp:124]
Jan 12 21:45:06 de838cd8-raspberrymatic user.warn monit[1470]: 'HMIPServer' failed protocol test [DEFAULT] at [localhost]:39292 [TCP/IP] -- Connection refused
Jan 12 21:45:07 de838cd8-raspberrymatic user.warn monit[1470]: 'hasUSB' status failed (1) -- no output
Jan 12 21:45:07 de838cd8-raspberrymatic user.err monit[1470]: 'rpi4usb3Check' status failed (1) -- no output
Jan 12 21:45:07 de838cd8-raspberrymatic user.err monit[1470]: Filesystem '/media/usb1' not mounted
Jan 12 21:45:07 de838cd8-raspberrymatic user.err monit[1470]: 'usb1' unable to read filesystem '/media/usb1' state
Jan 12 21:45:07 de838cd8-raspberrymatic user.info monit[1470]: 'usb1' trying to restart
Jan 12 21:45:22 de838cd8-raspberrymatic user.warn monit[1470]: 'HMIPServer' failed protocol test [DEFAULT] at [localhost]:39292 [TCP/IP] -- Connection refused
Jan 12 21:45:23 de838cd8-raspberrymatic user.warn monit[1470]: 'hasUSB' status failed (1) -- no output
Jan 12 21:45:23 de838cd8-raspberrymatic user.err monit[1470]: Filesystem '/media/usb1' not mounted
Jan 12 21:45:23 de838cd8-raspberrymatic user.err monit[1470]: 'usb1' unable to read filesystem '/media/usb1' state
Jan 12 21:45:23 de838cd8-raspberrymatic user.info monit[1470]: 'usb1' trying to restart
Jan 12 21:45:38 de838cd8-raspberrymatic user.err monit[1470]: 'HMIPServer' failed protocol test [DEFAULT] at [localhost]:39292 [TCP/IP] -- Connection refused
Jan 12 21:45:38 de838cd8-raspberrymatic user.info monit[1470]: 'HMIPServer' trying to restart
Jan 12 21:45:38 de838cd8-raspberrymatic user.info monit[1470]: 'HMIPServer' restart: '/etc/init.d/S62HMServer restart'
Jan 12 21:46:08 de838cd8-raspberrymatic user.err monit[1470]: 'HMIPServer' failed to restart (exit status -1) -- '/etc/init.d/S62HMServer restart': Program timed out -- Stopping HMIPServer: OK Starting HMIPServer: ...............
Jan 12 21:46:09 de838cd8-raspberrymatic user.err monit[1470]: 'hasUSB' status failed (1) -- no output
Jan 12 21:46:24 de838cd8-raspberrymatic user.err monit[1470]: 'HMIPServer' service restarted 1 times within 1 cycles(s) - exec
Jan 12 21:46:24 de838cd8-raspberrymatic user.info monit[1470]: 'HMIPServer' exec: '/bin/triggerAlarm.tcl HMIPServer restarted WatchDog: hmipserver-restart true'
Jan 12 21:46:24 de838cd8-raspberrymatic user.info monit[1470]: 'HMIPServer' process is running after previous exec error (slow starting or manually recovered?)
Jan 12 21:46:24 de838cd8-raspberrymatic user.info monit[1470]: 'HMIPServer' process is running after previous restart timeout (manually recovered?)
Jan 12 21:46:24 de838cd8-raspberrymatic user.err monit[1470]: 'HMIPServer' failed protocol test [DEFAULT] at [localhost]:39292 [TCP/IP] -- Connection refused
Jan 12 21:46:24 de838cd8-raspberrymatic user.info monit[1470]: 'HMIPServer' trying to restart
Jan 12 21:46:24 de838cd8-raspberrymatic user.info monit[1470]: 'HMIPServer' restart: '/etc/init.d/S62HMServer restart'
Jan 12 21:46:54 de838cd8-raspberrymatic user.err monit[1470]: 'HMIPServer' failed to restart (exit status -1) -- '/etc/init.d/S62HMServer restart': Program timed out -- Stopping HMIPServer: OK Starting HMIPServer: ...............
Jan 12 21:47:09 de838cd8-raspberrymatic user.info monit[1470]: 'HMIPServer' process is running after previous exec error (slow starting or manually recovered?)
Jan 12 21:47:09 de838cd8-raspberrymatic user.err monit[1470]: 'HMIPServer' failed protocol test [DEFAULT] at [localhost]:39292 [TCP/IP] -- Connection refused
Jan 12 21:47:09 de838cd8-raspberrymatic user.info monit[1470]: 'HMIPServer' trying to restart
Jan 12 21:47:09 de838cd8-raspberrymatic user.info monit[1470]: 'HMIPServer' restart: '/etc/init.d/S62HMServer restart'
Jan 12 21:47:44 de838cd8-raspberrymatic user.err monit[1470]: 'HMIPServer' service restarted 1 times within 1 cycles(s) - exec
Jan 12 21:47:44 de838cd8-raspberrymatic user.info monit[1470]: 'HMIPServer' exec: '/bin/triggerAlarm.tcl HMIPServer restarted WatchDog: hmipserver-restart true'
Jan 12 21:47:44 de838cd8-raspberrymatic user.info monit[1470]: 'HMIPServer' process is running after previous restart timeout (manually recovered?)
Jan 12 21:47:44 de838cd8-raspberrymatic user.info monit[1470]: 'HMIPServer' connection succeeded to [localhost]:39292 [TCP/IP]


***** hmserver.log *****
2023-01-12 21:39:20,247 de.eq3.cbcs.server.core.vertx.KeyServerWorker ERROR [vert.x-worker-thread-1] Missing key server configuration parameter (Network.Key) for  mode: KEYSERVER_LOCAL 
2023-01-12 21:45:40,903 de.eq3.cbcs.server.core.vertx.KeyServerWorker ERROR [vert.x-worker-thread-1] Missing key server configuration parameter (Network.Key) for  mode: KEYSERVER_LOCAL 
2023-01-12 21:46:26,549 de.eq3.cbcs.server.core.vertx.KeyServerWorker ERROR [vert.x-worker-thread-2] Missing key server configuration parameter (Network.Key) for  mode: KEYSERVER_LOCAL 
2023-01-12 21:47:12,811 de.eq3.cbcs.server.core.vertx.KeyServerWorker ERROR [vert.x-worker-thread-2] Missing key server configuration parameter (Network.Key) for  mode: KEYSERVER_LOCAL 
```

Kann mir jemand hier helfen?

**Comment by jens-maus** [comment]
Zeig doch mal die Inhalte der `/var/log/boot.log` oder das Startlog des RaspberryMatic add-ons. Das sollte aussagekräftiger sein.

Abgesehen davon würde ich aber ohnehin bei der Konstellation die du da hardwaretechnisch hast (Proxmox auf DELL server, etc.) dazu raten doch besser RaspberryMatic als eigenständig VM innerhalb von Proxmox zu betreiben und jetzt nicht als Addon innerhalb einer HomeAssistantOS VM. Das ist IMHO etwas zuviel der Virtualisierungsebenen die es nur unnötig kompliziert machen.

**Reply by philippsandhaus** [reply]
Hallo Jens, danke für den Hinweis. Auf die Installation direkt in Proxmox bin ich gar nicht gekommen, das werde ich auf jeden Fall auch gleich noch testen. Trotzdem hier erst einmal das Boot-Log:
/var/log/boot.log:

```
root@de838cd8-raspberrymatic:~# cat /var/log/boot.log 
Mounting /data as /usr/local (Home Assistant Add-On): OK
Identifying host system: oci, OK
Initializing RTC Clock: onboard, OK
Running sysctl: OK
Checking for Factory Reset: not required
Checking for Backup Restore: not required
Initializing System: OK
Setup ca-certificates: OK
Starting logging: OK
Init onboard LEDs: init, OK
Starting irqbalance: OK
Starting iptables: OK
Starting network: eth0: link up, fixed, firewall, inet up, 172.30.33.3, OK
Identifying Homematic RF-Hardware: ....HmRF: HMIP-RFUSB/eQ-3 HmIP-RFUSB@usb-0000:01:1b.0-3, HmIP: HMIP-RFUSB/eQ-3 HmIP-RFUSB@usb-0000:01:1b.0-3, OK
Updating Homematic RF-Hardware: HMIP-RFUSB: 4.4.18, not necessary, OK
Starting hs485dLoader: disabled
Starting xinetd: OK
Starting eq3configd: OK
Starting lighttpd: OK
Starting ser2net: disabled
Starting ssdpd: OK
Starting ha-proxy: OK
Starting NUT services: disabled
Initializing Third-Party Addons: OK
Starting LGWFirmwareUpdate: ...OK
Setting LAN Gateway keys: OK
Starting hs485d: disabled
Starting multimacd: .......OK
Starting rfd: .OK
Starting HMIPServer: .......................................................................................................................................................ERROR
Starting ReGaHss: .OK
Starting CloudMatic: OK
Starting NeoServer: disabled
Starting Third-Party Addons: OK
Starting crond: OK
Setup onboard LEDs: booted, OK
Finished Boot: 3.65.11.20221218 (raspmatic_oci_amd64)
```

**Reply by jens-maus** [reply]
Das sieht sehr murky aus das boot.log. Das Funkmodul wird wohl i.O. eingebunden aber der HMIPServer will nicht starten. Müsste man sich mal im HMIPServer debug mode starten und schauen warum nicht. Ist das ein fabrikneues RaspberryMatic addon oder nachdem du versuchst hast ein Backup einer anderen zentrale einzuspielen?

**Reply by philippsandhaus** [reply]
Nein, das ist ein "fabrikneues" RaspberryMatic Addon ohne eingespieltes Backup. Ich habe nun nach deinem Vorschlag eine Installation direkt unter Proxmox nach [Anleitung](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Proxmox-VE) durchgeführt. Hier erhalte ich die genannten Fehler nicht mehr. Allerdings lassen sich nach ersten Tests keine Geräte anlernen. Ich versuche nun hier testweise einmal ein Backup wieder einzuspielen.

**Reply by jens-maus** [reply]
> Hier erhalte ich die genannten Fehler nicht mehr. Allerdings lassen sich nach ersten Tests keine Geräte anlernen.

Bitte nicht vergessen das HmIP-RFUSB bei einer solchen Nutzung so weit es geht via USB Verlängerungskabel und am besten auch mit einem USB Entstörfilter (https://de.elv.com/elv-usb-entstoerfilter-usb-ef1-komplettbausatz-152745) vom Virtualisierungshost abzusetzen. Gerade für den HmIP Funk benötigt das Funkmodul eine Störungsfreie RF-Umgebung und ein USB3 Anschluss, sowie die zu große Nähe zu anderen funkenden Geräten ist absolut schädlich und erklärt auch warum man nichts anlernen kann. Deshalb 2-3 m gut geschirmtes USB Verlängerungskabel samt USB Entstörfilter wirken mitunter wunder. Das gilt bei Einsatz als HA Addon oder als Standalone RaspberryMatic oder als RaspberryMatic VM genauso.

**Reply by philippsandhaus** [reply]
Ok, dann werde ich mal weiter in diese Richtung experimentieren. Der Stick ist im Moment mit einem (zumindest denke ich das das) guten USB Verlängerungskabel an geschlossen, allerdings nur 1m und ohne Entstörfilter. Auch verschiedene Positionen bringen nach ersten Testen nichts. Danke erst einmal für die Hinweise!

**Reply by djtale2003** [reply]
Muss der Stick an einen USB3 Port, sowohl das beiliegende Kabel als auch der Stecker ist doch USB2? Ich habe RM auf Proxmox installiert. Wenn ich die Netzwerkkarte entferne, startet der HMIP-Server, mit Netzwerkkarte bekomme ich immer ERROR...

Update:
Ich habe es an einem USB3 Port versucht, gleiches Ergebniss: HMIP-Server -> ERROR

**Reply by emkaaay** [reply]
Hi, ich hab ähnliches problem, habe das Add-on in HA laufen und hab es mit dem HMIP-RFUSB connected, der USB stick wurde auch als neues Gerät angelernt/erkannt. Anschliessend hab ich mein backup vom pi eingespielt und sehe auch wunderbar alle Geräte und Programme. Allerdings steht bei allen "Gerätekommunikation gestört", evtl hat ja jemand noch eine Idee... Würde ungern alle Geräte am Pi ablernen um zu versuchen die im HA Addon neu anzulernen. Hab das RaspberryMatic Log von HA auch mal hier eingefügt, sieht sehr identisch zu deinem aus wenn ich das richtig sehe... Hab den stick übrigens auch am USB2.0 port laufen mit dem beigefügten Verlängerungskabel von ELV. Hardware bei mir ist ein intel nuc i5.

`Mounting /data as /usr/local (Home Assistant Add-On): OK
Identifying host system: Intel Corporation NUC7i5DNB (oci), OK
Initializing RTC Clock: onboard, OK
Running sysctl: OK
Checking for Factory Reset: not required
Checking for Backup Restore: not required
Running seedrng: OK
Initializing System: OK
Setup ca-certificates: OK
Starting logging: OK
Init onboard LEDs: init, OK
Starting iptables: OK
Starting network: eth0: link up, fixed, firewall, inet up, 172.30.33.1, OK
Identifying Homematic RF-Hardware: ....HmRF: HMIP-RFUSB/eQ-3 HmIP-RFUSB@usb-0000:00:14.0-4, HmIP: HMIP-RFUSB/eQ-3 HmIP-RFUSB@usb-0000:00:14.0-4, OK
Updating Homematic RF-Hardware: HMIP-RFUSB: 4.4.18, not necessary, OK
Starting hs485dLoader: disabled
Starting xinetd: OK
Starting eq3configd: OK
Starting lighttpd: OK
Starting ser2net: disabled
Starting ssdpd: OK
Starting ha-proxy: OK
Starting NUT services: disabled
Initializing Third-Party Addons: OK
Starting LGWFirmwareUpdate: ...OK
Setting LAN Gateway keys: OK
Starting hs485d: disabled
Starting multimacd: .......OK
Starting rfd: .OK
Starting HMIPServer: ........OK
Starting ReGaHss: .OK
Starting CloudMatic: OK
Starting NeoServer: OK
Starting Third-Party Addons: OK
Starting crond: OK
Setup onboard LEDs: booted, OK
Finished Boot: 3.75.7.20240420 (raspmatic_oci_amd64)`

**Reply by emkaaay** [reply]
hab hier auch nochmal das diagnose log aus HA für die raspberry integration, kan den USB stick auch als gerät dort finden.
`{
  "home_assistant": {
    "installation_type": "Home Assistant OS",
    "version": "2024.4.3",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.12.2",
    "docker": true,
    "arch": "x86_64",
    "timezone": "Europe/Berlin",
    "os_name": "Linux",
    "os_version": "6.6.25-haos",
    "supervisor": "2024.04.0",
    "host_os": "Home Assistant OS 12.2",
    "docker_version": "25.0.5",
    "chassis": "embedded",
    "run_as_root": true
  },
  "custom_components": {
    "hacs": {
      "version": "1.34.0",
      "requirements": [
        "aiogithubapi>=22.10.1"
      ]
    },
    "homematicip_local": {
      "version": "1.59.0",
      "requirements": [
        "hahomematic==2024.4.6"
      ]
    },
    "gardena_smart_system": {
      "version": "1.0.0",
      "requirements": [
        "py-smart-gardena==1.3.8"
      ]
    }
  },
  "integration_manifest": {
    "domain": "homematicip_local",
    "name": "Homematic(IP) Local",
    "codeowners": [
      "@danielperna84",
      "@SukramJ"
    ],
    "config_flow": true,
    "dependencies": [],
    "documentation": "https://github.com/danielperna84/custom_homematic",
    "homekit": {},
    "integration_type": "hub",
    "iot_class": "local_push",
    "issue_tracker": "https://github.com/danielperna84/hahomematic/issues",
    "loggers": [
      "hahomematic"
    ],
    "requirements": [
      "hahomematic==2024.4.6"
    ],
    "ssdp": [
      {
        "manufacturer": "EQ3",
        "manufacturerURL": "http://www.homematic.com"
      }
    ],
    "version": "1.59.0",
    "zeroconf": [],
    "is_built_in": false
  },
  "data": {
    "config": {
      "entry_id": "52e40ff515124fedc5ba876ef7c6332c",
      "version": 2,
      "minor_version": 1,
      "domain": "homematicip_local",
      "title": "RaspberryMatic",
      "data": {
        "instance_name": "RaspberryMatic",
        "host": "de838cd8-raspberrymatic",
        "username": "**REDACTED**",
        "password": "**REDACTED**",
        "tls": false,
        "verify_tls": false,
        "sysvar_scan_enabled": true,
        "sysvar_scan_interval": 30,
        "callback_host": null,
        "callback_port": null,
        "json_port": null,
        "enable_system_notifications": true,
        "interface": {
          "HmIP-RF": {
            "port": 2010
          },
          "BidCos-RF": {
            "port": 2001
          },
          "VirtualDevices": {
            "port": 9292,
            "path": "/groups"
          }
        }
      },
      "options": {},
      "pref_disable_new_entities": false,
      "pref_disable_polling": false,
      "source": "ssdp",
      "unique_id": "22698D6843",
      "disabled_by": null
    },
    "platform_stats": {
      "binary_sensor": 24,
      "climate": 11,
      "cover": 14,
      "hub_binary_sensor": 12,
      "hub_button": 42,
      "hub_sensor": 50,
      "light": 2,
      "sensor": 7,
      "switch": 15,
      "update": 44
    },
    "devices": [
      "HM-CC-RT-DN",
      "HM-CC-VG-1",
      "HM-LC-Bl1PBU-FM",
      "HM-LC-Dim1TPBU-FM",
      "HM-LC-Sw1-FM",
      "HM-LC-Sw2-FM",
      "HM-LC-Sw4-PCB",
      "HM-RCV-50",
      "HM-Sen-Wa-Od",
      "HM-TC-IT-WM-W-EU",
      "HMIP-PS",
      "HMIP-SWDO",
      "HmIP-FBL",
      "HmIP-RCV-50",
      "HmIP-RFUSB",
      "HmIP-SLO",
      "HmIP-STHO-A",
      "HmIP-SWSD"
    ],
    "system_information": {
      "available_interfaces": [
        "BidCos-RF",
        "VirtualDevices",
        "HmIP-RF"
      ],
      "auth_enabled": false,
      "https_redirect_enabled": false,
      "serial": "**REDACTED**"
    }
  }
}`
